indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -422,6 +428,69 @@ public Response getWorkerConfig()
     return Response.ok(workerConfigRef.get()).build();
   }
 
+  /**
+   * Gets the total worker capacity of varies states of the cluster.
+   */
+  @GET
+  @Path("/totalWorkerCapacity")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(ConfigResourceFilter.class)
+  public Response getTotalWorkerCapacity()
+  {
+    // Calculate current cluster capacity
+    int currentCapacity;
+    Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
+    if (!taskRunnerOptional.isPresent()) {
+      // Cannot serve call as not leader
+      return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
```

Comment:
nit: log line to help with debugging
```suggestion
      log.info("Can not calculate total worker capacity as TaskRunner is not present.");
      return Response.status(Response.Status.SERVICE_UNAVAILABLE).build();
```